onnx/test/shape_inference_test.py

```diff
@@ -1667,6 +1702,38 @@ def test_if(self):  # type: () -> None
 
         self._assert_inferred(graph, [make_tensor_value_info('if_output', TensorProto.FLOAT, (1,))])
 
+    def test_if_with_different_shapes_in_then_else_branches(self):  # type: () -> None
+
+        # Create a simple If node where the 'then' subgraph adds to the current value, and the 'else' subgraph
+        # subtracts.
+        # can't use self._make_graph for the subgraphs as that add more inputs for the Reshape operations it inserts.
```

Comment:
Wording is awkward here, perhaps `avoid calling self._make_graph here for subgraph generation as extra reshape nodes will be created`

Reply:
I feel the current description is more precise. From this description, adding Reshape itself is not a problem, but adding input required by Reshape causes the problem.